Yoast/YoastSEO.js
# Refactor keyword density to include morphology

## Summary

This PR can be summarized in the following changelog entry:

* The keyword density assessment now takes all possible morphological forms of the keywords into account.

## Relevant technical choices:

* An occurrence is counted when all the keywords are contained within one sentence.
    * E.g. "An apple and a potato." is counted as one occurrence for the key phrase [potato apple]
* Multiple occurrences of a key phrase within a sentence are counted.
    * E.g. "A potato apple is an apple as well as a potato." is counted as two occurrences. 

## Test instructions

This PR can be tested by following these steps:

* Make a new post.
* Add text of 150 words to trigger the keyword density assessment ([e.g use this Lorem Ipsum generator](https://lipsum.com/)).

Test these test cases:
* Text containing none of the keywords --> 0% keyword density.
* Text containing one sentence with all the keywords once --> 0.7% keyword density.
* Text containing one sentence with all the keywords twice --> 1.3% keyword density.
* Text containing two sentences with all the keywords once --> 1.3% keyword density.
* Text containing two sentence with all the keywords twice --> 2.6% keyword density.
* Make sure that markings work and that all conditions have the new feedback strings with working links.

Fixes #1785
